system/security/plugins/singleuserSecurity/singleUserSecurity.cpp

```diff
@@ -0,0 +1,216 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma warning( disable : 4786 )
+
+#include "basesecurity.hpp"
+#include "authmap.ipp"
+#include "singleUserSecurity.hpp"
+
+class CSingleUserSecurityManager : public CBaseSecurityManager
+{
+public:
+    CSingleUserSecurityManager(const char *serviceName, IPropertyTree *secMgrCfg, IPropertyTree *bindConfig) : CBaseSecurityManager(serviceName, (IPropertyTree *)NULL)
+    {
+        if (secMgrCfg)
```

Comment:
Should you throw exception if secMgrCfg is nullptr? If secMgrCfg is nullptr, m_userPass will not be set and the authorize() will throw exception. So, why not throw exception here if secMgrCfg is nullptr?

Reply:
I agree, the problem is we tend not to throw in a constructor